decred/decred/dcr/dcrdata.py

```diff
@@ -665,29 +666,14 @@ def blockForTx(self, txid):
         except database.NoValueError:
             # If the blockhash is not in the database, get it from dcrdata
             decodedTx = self.dcrdata.tx(txid)
-            if (
-                "block" not in decodedTx
-                or "blockhash" not in decodedTx["block"]
-                or decodedTx["block"]["blockhash"] == ""
-            ):
+            try:
+                hexHash = decodedTx["block"]["blockhash"]
+            except KeyError:
```

Comment:
This is not equivalent. The check for empty block hash is critical.

```
2020-03-02 11:46:46,356 dcrdata WARNING blockHeader(697) unable to retrieve block header: ('RequestError', 'Error encountered in requesting path http://localhost:17778/api/block/hash//header/raw: HTTP Error 422: Unprocessable Entity\nNone')

```

Reply:
Oops, mixed the last line, fixing.